nextroute/options.go

```diff
@@ -0,0 +1,42 @@
+package nextroute
+
+import (
+	"github.com/nextmv-io/sdk/nextroute/schema"
+)
+
+// An Option configures a model.
+type Option func(Model, schema.Input) error
+
+// ModelOptions configures how a [Model] is built.
+type ModelOptions struct {
```

Comment:
these should be called FactoryOptions and you can opt to call it just Options as it will move to the factory package